query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -35,13 +43,25 @@ final case class PeriodicSamplesMapper(start: Long,
             queryConfig: QueryConfig,
             limit: Int,
             sourceSchema: ResultSchema): Observable[RangeVector] = {
-    RangeVectorTransformer.requireTimeSeries(sourceSchema)
-    source.map { rv =>
-      IteratorBackedRangeVector(rv.key,
-        new SlidingWindowIterator(rv.rows, start, step, end, window.getOrElse(0L),
-          RangeFunction(functionId, funcParams), queryConfig))
+    RangeVectorTransformer.valueColumnType(sourceSchema) match {
+      case ColumnType.DoubleColumn =>
+        source.map { rv =>
+          IteratorBackedRangeVector(rv.key,
+            new SlidingWindowIterator(rv.rows, start, step, end, window.getOrElse(0L),
+              RangeFunction(functionId, funcParams), queryConfig))
+        }
+      case ColumnType.LongColumn =>
+        source.map { rv =>
+          IteratorBackedRangeVector(rv.key,
+            new SlidingWindowIterator(new LongToDoubleIterator(rv.rows), start, step, end, window.getOrElse(0L),
```

Comment:
It seems the bulk of this change is to convert longs to doubles automatically. That might cause a surprise if someone wanted the extra precision. When would a column type be defined as a long if only to convert it to double later?

Reply:
Most of the range functions are going to be converted to type specific aggregations, but this will take time.  I agree the current state of this change doesn't offer any precision advantages, but it does unblock until over time we can add in the necessary query functionality.